pygmt/clib/session.py

```diff
@@ -1946,6 +1951,70 @@ def virtualfile_to_dataset(
             return result.to_numpy()
         return result  # pandas.DataFrame output
 
+    def virtualfile_to_raster(
```

Comment:
> Most GMT modules can write grids only, so we can set `kind="grid"` to avoid calling `Session.inquire_virtualfile`.

Yes, I think we can set `kind="grid"`, unless we know if GMT will allow passing GMT_IMAGE to certain grd* modules in the future. For example, maybe `grdclip` would allow GMT_IMAGE inputs at some point, and so we might want to set `kind=None` for future-proofing.

Reply:
We need to do more refactoring if `grdclip` would support GMT_IMAGE in the future, since we need to check the kind of the input data (grid in, grid out or image in, image out), similar to what we have to do to `grdcut` (xref: https://github.com/GenericMappingTools/pygmt/pull/2398#discussion_r1535234139).

Reply:
Ok, I meant if upstream GMT were to support passing `GMT_IMAGE` into `grdclip` in the future, but not sure when that would happen. Ok to just default to `kind='grid'` for now, since that's the current behaviour I think.